### PR TITLE
Refactor serializer factory logic to avoid passing many out-variables to functions

### PR DESCRIPTION
**Pull request by vdboor**
Instead collect everything in a single SerializerAssemblyLine class.
This also streamlines how both 'fields' and 'new_attrs' has to be set.

### CODE REVIEW

File: src/dso_api/dynamic_api/serializers.py [hunk 1]
```diff
@@ -512,26 +558,23 @@ def serializer_factory(
...
-        "__module__": f"dso_api.dynamic_api.serializers.{safe_dataset_id}",
-        "__doc__": table_schema.get(
-            "description", table_schema.name
-        ),  # avoid exposing our docstrings.
-    }
+
+    serializer_part = SerializerAssemblyLine(model, depth=depth)
+
+    # Generate a proper docstring for the OpenAPI spec
+    serializer_part.class_attrs["__doc__"] = table_schema.get("description", table_schema.name)
```
Comment (jjmurre) [inline src/dso_api/dynamic_api/serializers.py:571]: Why not doing this inside `SerializerAssemblyLine`? I saw the comment about not exposing the docstring. Does doing it like this avoid exposing the docstring?

**Review by jjmurre — APPROVED** [review]
Nice refactoring.

File: src/dso_api/dynamic_api/serializers.py (hunk 1 above)
Comment (vdboor) [reply src/dso_api/dynamic_api/serializers.py:571]: That class is reused for more serializer construction, including the `_links` serializer and through serializer. I didn't want to render those docs for these classes as well.

Comment (vdboor) [reply src/dso_api/dynamic_api/serializers.py]: I've moved this to a constructor argument

**Pull request merged**